app/models/build_queue.rb

```diff
@@ -0,0 +1,42 @@
+# == Schema Information
+#
+# Table name: build_queues
+#
+#  id           :uuid             not null, primary key
+#  applied_at   :datetime
+#  is_active    :boolean          default(TRUE)
+#  scheduled_at :datetime         not null
+#  created_at   :datetime         not null
+#  updated_at   :datetime         not null
+#  release_id   :uuid             not null, indexed
+#
+class BuildQueue < ApplicationRecord
+  belongs_to :release, inverse_of: :build_queues
+  has_many :commits, dependent: :destroy, inverse_of: :build_queue
+
+  scope :active, -> { where(is_active: true) }
+  delegate :train, to: :release
+
+  after_create_commit :schedule_kickoff!
+
+  def apply!
```

Comment:
I think we can subsume these checks:

```ruby
return unless build_queue.release.committable?
return unless build_queue.is_active?
```

within the `apply` itself, wrap it in Result object and have the caller parse.

Additionally, we can also optionally take a lock on the release with an argument like `requires_lock: true`.

Reply:
This may end up doing a `release.committable?` check twice, but it's cheap enough that it shouldn't matter.